mod/01_rating/set/type/metric/creation.rb

```diff
@@ -32,17 +32,31 @@ def add_value_source_args args, source
   end
 end
 
-def valid_value_args? args
-  error_msg = []
-  metric_value_name =
-    args[:name] || begin
-      missing = [:company, :year, :value].reject { |v| args[v] }
-      error_msg += missing.map { |field| "missing #{field}" }
+def extract_metric_value_name args, error_msg
+  args[:name] || begin
+    missing = [:company, :year, :value].reject { |v| args[v] }
+    if missing.empty?
       [name, args[:company], args[:year]].join "+"
+    else
+      error_msg.push("missing field(s) #{missing.join(',')}")
+      nil
+    end
+  end
+end
+
+def check_value_card_exist args, error_msg
+  return unless (value_name = extract_metric_value_name(args, error_msg))
+  if (value_card = Card[value_name.to_name.field(:value)])
+    unless value_card.content.casecmp(args[:value]) == 0
```

Comment:
inconsistent style here.  `return unless`, `if`, and `unless` are all similar.  would read more cleanly if it were something like:

```
return unless (value_name = extract_metric_value_name args, error_msg)
return unless (value_card = Card[value_name.to_name.field :value])
return if value_card.content.casecmp(args[:value]) == 0
...
```